### PR TITLE
Fix #187: ModuleNotFoundError: No module named 'django' when using SQLAlchemy

### LINKED ISSUE #187: ModuleNotFoundError: No module named 'django' when using SQLAlchemy
Looks like when trying to use SQLAlchemy (and not Django ORM) in master, we get a failure:

```bash
$ apistar run
Traceback (most recent call last):
  File "/Users/jriggins/src/optimumsymmetry/survey-system-2/python/venv/bin/apistar", line 11, in <module>
    load_entry_point('apistar==0.1.17', 'console_scripts', 'apistar')()
  File "/Users/jriggins/src/optimumsymmetry/survey-system-2/python/venv/lib/python3.6/site-packages/apistar/cli.py", line 44, in main
    app = get_current_app()
  File "/Users/jriggins/src/optimumsymmetry/survey-system-2/python/venv/lib/python3.6/site-packages/apistar/cli.py", line 27, in get_current_app
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/Users/jriggins/src/optimumsymmetry/survey-system-2/python/app.py", line 5, in <module>
    from apistar.backends.sqlalchemy_backend import SQLAlchemy
  File "/Users/jriggins/src/optimumsymmetry/survey-system-2/python/venv/lib/python3.6/site-packages/apistar/backends/__init__.py", line 1, in <module>
    from apistar.backends.django_backend import DjangoBackend
  File "/Users/jriggins/src/optimumsymmetry/survey-system-2/python/venv/lib/python3.6/site-packages/apistar/backends/django_backend.py", line 1, in <module>
    from django import setup
ModuleNotFoundError: No module named 'django'
```

### PR DESCRIPTION
**Pull request by jriggins**
Fix for #187.  I'm sure that there is a more elegant way to fix this, but it seems to have gotten the job done.

### CODE REVIEW

**Comment by audiolion** [comment]
I just issued a PR for [rest framework](https://github.com/encode/django-rest-framework/pull/5185) that fixes the issue, the same method is used here that is breaking and we could probably do the same solution

**Comment by jriggins** [comment]
@audiolion  That sounds good.  I had not looked into actually _fixing_ the requests issue.  😄   Regardless of my bandaid (which did get around the requests issue), it still looks like Travis CI builds are still failing.  I'm still new to all of this, but looks like the tests pass but the script exits with a nonzero status.

**Comment by audiolion** [comment]
No worries, your CI is almost passing! Just a few lint errors

```
apistar/backends/__init__.py:4:5: F401 'apistar.backends.django_backend.DjangoBackend' imported but unused
apistar/backends/__init__.py:10:5: F401 'apistar.backends.sqlalchemy_backend.SQLAlchemy' imported but unused
apistar/backends/__init__.py:14:1: W391 blank line at end of file
```

You can run `scripts/lint` before pushing and it will tell you of these issues. In the case of unused imports you remove them, and you have an extra blank line at the end of the __init__ file

**Comment by jriggins** [comment]
Thanks @audiolion

**Comment by jriggins** [comment]
I'm wondering in the file that I was trying to fix whether I should be passing ```# noqa``` for the imports similar to this [SO](https://stackoverflow.com/questions/31079047/python-pep8-class-in-init-imported-but-not-used)

File: apistar/backends/__init__.py [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+    from apistar.backends.django_backend import DjangoBackend  # noqa
+    __all__.append('DjangoBackend')
+except ImportError:
+    pass
+
+try:
+    from apistar.backends.sqlalchemy_backend import SQLAlchemy  # noqa
+    __all__.append('SQLAlchemy')
+except ImportError:
+    pass
```
Comment (tomchristie) [inline apistar/backends/__init__.py]: Have a look at the `compat.py` module, and follow the same style if possible. 👍

**Comment by tomchristie** [comment]
We no longer need the django/sqlalchemy part of this.

I'm good with us switching to using `urllib3` directly. We could list a minimum version for requests in our setup.py rather than using `compat.py`.

**Pull request closed without merging**